frontend/src/queries/QueryEditor/QueryEditor.tsx

```diff
@@ -0,0 +1,63 @@
+import { useActions, useValues } from 'kea'
+import MonacoEditor, { useMonaco } from '@monaco-editor/react'
+import { useEffect, useState } from 'react'
+import schema from '~/queries/schema.json'
+import { LemonButton } from 'lib/components/LemonButton'
+import { queryEditorLogic } from '~/queries/QueryEditor/queryEditorLogic'
+
+export interface QueryEditorProps {
+    query: string
+    setQuery?: (query: string) => void
+}
+
+let i = 0
+export function QueryEditor(props: QueryEditorProps): JSX.Element {
+    const [key] = useState(() => i++)
+    const { queryInput, error, inputChanged } = useValues(queryEditorLogic({ ...props, key }))
+    const { setQueryInput, saveQuery } = useActions(queryEditorLogic({ ...props, key }))
+    const monaco = useMonaco()
+
+    useEffect(() => {
+        if (!monaco) {
+            return
+        }
+        monaco.languages.json.jsonDefaults.setDiagnosticsOptions({
+            validate: true,
+            schemas: [
+                {
+                    uri: 'https://internal.posthog.com/node-schema.json',
```

Comment:
Let's make it clear that this isn't a real URI/is not used.

Reply:
I put `http://internal/node-schema.json`. Hopefully that's internal enough 🙃